src/rattler_build_conda_compat/render.py

```diff
@@ -59,10 +64,53 @@ def __init__(
 
         self.requirements_path = os.path.join(self.path, "requirements.txt")
 
-    def parse_recipe(self):
-        yaml = YAML()
-        with open(os.path.join(self.path, self._meta_name), "r") as recipe_yaml:
-            return yaml.load(recipe_yaml)
+    def parse_recipe(self) -> dict[str, Any]:
+        recipe_path: Path = Path(self.path) / self._meta_name
+
+        yaml_content = load_yaml(recipe_path.read_text())
+
+        return render_recipe_with_context(yaml_content)
+
+    def name(self) -> str:
```

Comment:
this override name() and version() from conda_build.Metadata base class. typing override was introduced in 3.12 so I cannot use it here